public/app/plugins/datasource/loki/types.ts

```diff
@@ -10,6 +10,8 @@ export interface LokiQuery extends DataQuery {
 
 export interface LokiOptions extends DataSourceJsonData {
   maxLines?: string;
+  derivedFields?: DerivedFieldConfig[];
+  dataLinks?: DataLink[];
```

Comment:
Hmm the structure to match the fields is the:
```
derivedFields?: DerivedFieldConfig[];
```
So that can be part of the dataframe somewhere and done during querying (the only problem with this thought that we will need to filter them somehow from table views as they will show them by default and they may not be meaningful without the links).

Regarding the links I am not sure about the logic of making them part of the dataframe. Right now the context of the datalinks template can be very custom and could take variables from outside of the data frame so the relation of them does not seem that clear to me. Like for example I think the context can also be dashboard variables. Also they are part of panel config and now datasource config. In that sense having them as part of dataframe seems strange to me.